satpy/scene.py

```diff
@@ -41,6 +41,26 @@
 LOG = logging.getLogger(__name__)
 
 
+def _get_area_resolution(area):
+    """Attempt to retrieve resolution from AreaDefinition."""
+    try:
+        resolution = max(area.pixel_size_x, area.pixel_size_y)
+    except AttributeError:
+        resolution = max(area.lats.resolution, area.lons.resolution)
+    return resolution
```

Comment:
This feels like it should already be present somewhere in pyresample. I think we have a method called `geocentric_resolution`, can it be used here?

Reply:
It's not the same thing. [`geocentric_resolution`] (https://github.com/pytroll/pyresample/blob/12892b875221846f0fbc7f68775dbdadd8b68584/pyresample/geometry.py#L661) does not use the `pixel_size_x/y` attribute for AreaDefinition. 

To me, it seems that `geocentric_resolution` is more designed for SwathDefinition, so I would rather use it in place of  `resolution = max(area.lats.resolution, area.lons.resolution)` ... but not sure is the correct behavior ... 
Maybe would be worth in a separate PR to add an attribute method `resolution` for SwathDefinition and AreaDefinition and delegate to this all the logics

Reply:
I could do that but it would break the back compatibility ! 
For AreaDefinition for example, right now (and in the past) the method is/was using pixel_size (in units of projection coordinates). While `geocentric_resolution` if I understand correctly would provides the geocentric resolution (in degrees)

Reply:
You are correct that it is not the same, but I do want to clarify it should be in meters...just not the same meters as the projection. I think, especially since you didn't write this code, that it can stay as is (again, my opinion). I do think the `.resolution` should be `.attrs["resolution"]` though. If you have other changes to make maybe you could do that too.

Reply:
No no, sorry. I mean in the except clause it does `area.lats.resolution`. If I'm understanding this code `area` is a `SwathDefinition` in this `except` block, `.lats` is a `DataArray`, and `area.lats.resolution` is a shortcut to `area.lats.attrs["resolution"]`. The `.attrs` form of accessing this information is the preferred way of doing it.

Reply:
Done and likely ready to be merged ;)